File: app/src/main/java/ru/systempla/weatherapp/mvp/model/settings/PaperBasedSetting.java
package ru.systempla.weatherapp.mvp.model.settings;

import io.paperdb.Paper;
import io.reactivex.Completable;
import io.reactivex.Single;

public class PaperBasedSetting implements ISettings {

    @Override
    public void saveSetting(String setting) {
        Completable.fromAction(()-> Paper.book("settings").write("setting",setting)).subscribe();
    }

    @Override
    public Single<String> getSetting() {
        return Single.fromCallable(()->Paper.book("settings").read("setting"));
    }

    @Override
    public void resetSetting() {
        Completable.fromAction(()->Paper.book("settings").write("setting","gps")).subscribe();
    }
}

File: app/src/main/java/ru/systempla/weatherapp/mvp/model/settings/PaperBasedSetting.kt
package ru.systempla.weatherapp.mvp.model.settings

import io.paperdb.Paper
import io.reactivex.Completable
import io.reactivex.Single

class PaperBasedSetting : ISettings {
    override fun saveCitySetting(setting: String) {
        Completable.fromAction { Paper.book("settings").write("setting", setting) }.subscribe()
    }

    override val citySetting: Single<String>
        get() = Single.fromCallable<String> { Paper.book("settings").read("setting") }

    override fun resetCitySetting() {
        Completable.fromAction { Paper.book("settings").write("setting", "gps") }.subscribe()
    }

    override fun savePermissionsSetting(setting: Int) {
        Completable.fromAction { Paper.book("settings").write("permissions", setting) }.subscribe()
    }

    override val permissionsSetting: Single<Int>
        get() = Single.fromCallable<Int> { Paper.book("settings").read("permissions") }

    override fun resetPermissionsSetting() {
        Completable.fromAction { Paper.book("settings").write("permissions", 1) }.subscribe()
    }
}